rdkit/Chem/Fingerprints/FingerprintMols.py

```diff
@@ -326,25 +321,19 @@ def _clusterInit(self):
     self.actName = ''
 
   def GetMetricName(self):
-    if self.metric == DataStructs.TanimotoSimilarity:
-      return 'Tanimoto'
-    elif self.metric == DataStructs.DiceSimilarity:
-      return 'Dice'
-    elif self.metric == DataStructs.CosineSimilarity:
-      return 'Cosine'
-    elif self.metric:
-      return self.metric
-    else:
+    metricDict = {DataStructs.DiceSimilarity: 'Dice', DataStructs.TanimotoSimilarity: 'Tanimoto', 
+                  DataStructs.CosineSimilarity: 'Cosine', } # DataStructs.TverskySimilarity: 'Tversky'
+    try:
+      return metricDict[self.metric]
+    except KeyError:
+      if self.metric:
+        return self.metric
       return 'Unknown'
 
   def SetMetricFromName(self, name):
-    name = name.upper()
-    if name == "TANIMOTO":
-      self.metric = DataStructs.TanimotoSimilarity
-    elif name == "DICE":
-      self.metric = DataStructs.DiceSimilarity
-    elif name == "COSINE":
-      self.metric = DataStructs.CosineSimilarity
+    metricDict = {'DICE': DataStructs.DiceSimilarity, 'TANIMOTO': DataStructs.TanimotoSimilarity, 
+                  'COSINE': DataStructs.CosineSimilarity, } # 'TVERSKY': DataStructs.TverskySimilarity, 
+    self.metric = metricDict[name.upper()]
```

Comment:
The function now fails if name isn't found in the dictionary. Use
```
Self.metric = metricDict.get(name.upper())
```
It actually should be
```
Self.metric = metricDict.get(self.metric,self.metric)
```
As in the original code self.metric isn't changed.

Reply:
`self.metric = metricDict.get(name.upper(),self.metric)`